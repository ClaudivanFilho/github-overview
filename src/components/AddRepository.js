import React, { Component } from 'react'
import PropTypes from 'prop-types'
import ToastContext from '../contexts/ToastContext'
import RepositoryContext from '../contexts/RepositoryContext'
import AddIcon from '../images/add.svg'
import Api from '../api'

class AddRepository extends Component {
  static propTypes = {
    openToast: PropTypes.func,
    refetchRepositories: PropTypes.func,
  }

  state = {
    repInput: '',
  }

  handleSubmit = () => {
    if (!this.state.repInput.match(/.*\/.*/)) {
      this.props.openToast('Repository name must match with the pattern "owner/name"', true)
      return
    }
    const [owner, name] = this.state.repInput.split('/')
    Api.addRepository(owner, name)
      .then(res => {
        if (res.error) {
          this.props.openToast(res.error, true)
        } else {
          this.props.refetchRepositories()
          this.props.openToast(res.message)
        }
      })
      .catch(res => this.props.openToast(res.error, true))
  }

  render() {
    return (
      <div className="flex items-center">
        <input
          id="repository-name"
          className="input-reset ba b--black-20 db w-auto pa0 pl2 bg-white br2"
          placeholder="Owner/Name"

          title="Add Repository"
          value={this.state.repInput}
          onChange={evt => this.setState({ repInput: evt.target.value })}
        />
        <img
          className="flex items-center br7 ml2 pointer dim"
          src={AddIcon} alt="add-icon"
          width="25"
          onClick={this.handleSubmit}
        />
      </div>
    )
  }
}

export default function ComponentWithContext(props) {
  return (
    <ToastContext.Consumer>
      {({ openToast }) => (
        <RepositoryContext.Consumer>
          {({ refetchRepositories }) => (
            <AddRepository
              {...props}
              openToast={openToast}
              refetchRepositories={refetchRepositories} />
          )}
        </RepositoryContext.Consumer>
      )}
    </ToastContext.Consumer>
  )
}
