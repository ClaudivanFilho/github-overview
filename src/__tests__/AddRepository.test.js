import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import AddRepository from '../components/AddRepository'

describe('Test AddRepository component', () => {
  let mountedComponent = null

  beforeAll(() => {
    mountedComponent = mount(
      <AddRepository />
    )
  })

  it('Renders correctly', () => {
    expect(toJson(mountedComponent)).toMatchSnapshot()
  })

  it('Component should be defined', () => {
    expect(mountedComponent).toBeDefined()
  })

  it('Must find an input component', () => {
    expect(mountedComponent.find('input').exists()).toBe(true)
  })

  it('Must find the add icon', () => {
    expect(mountedComponent.find('img').exists()).toBe(true)
  })
})
