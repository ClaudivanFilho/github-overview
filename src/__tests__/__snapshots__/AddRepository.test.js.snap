// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddRepository component Renders correctly 1`] = `
<ComponentWithContext>
  <AddRepository
    openToast={[Function]}
    refetchRepositories={[Function]}
  >
    <div
      className="flex items-center"
    >
      <input
        className="input-reset ba b--black-20 db w-auto pa0 pl2 bg-white br2"
        id="repository-name"
        onChange={[Function]}
        placeholder="Owner/Name"
        title="Add Repository"
        value=""
      />
      <img
        alt="add-icon"
        className="flex items-center br7 ml2 pointer dim"
        onClick={[Function]}
        src="add.svg"
        width="25"
      />
    </div>
  </AddRepository>
</ComponentWithContext>
`;
